ref/anbox-https-api.md

```diff
@@ -491,3 +494,211 @@ Platform | Field name       | Available since   | JSON type | Access | Descripti
 ---------|------------------|-------------------|-----------|--------|--------------------|
 `webrtc` | `rtc_log`         | 1.15 | Boolean   | read/write | Enable/disable [RTC event logging](https://webrtc.googlesource.com/src/+/lkgr/logging/g3doc/rtc_event_log.md). Logs are written to `/var/lib/anbox/traces/rtc_log.*` inside the instance. |
 `webrtc` | `stream_active`   | 1.15 | Boolean   | read | `true` if a client is actively streaming, `false` if no client is connected. |
+
+
+<a name="heading--10vhal"></a>
+### `/1.0/vhal`
+
+This endpoint queries the [Android VHAL](https://source.android.com/docs/automotive/vhal)
+through Anbox. It mimics the
+[VHAL HIDL interface](https://source.android.com/docs/automotive/vhal/hidl-vhal-interface)
+for `get` and `set` and follows RESTful API conventions. All queries on this
+endpoint will fail with a 500 error code on non-automotive Anbox images.
+
+#### GET `1.0/vhal/{prop_id}/{area_id}`
+ * Description: Get a VHAL property value
+ * Operation: sync
+ * Return: Current value for requested property
+ * Parameters:
+    * `prop_id`: Property identifier. Can be given in decimal, octal or hexadecimal format.
+    * `area_id`: Valid area identifier for the property. Can be omitted for global properties. Can be given in decimal, octal, or hexadecimal format.
+
+To get the list of available properties and areas, query first the [`1.0/vhal/config` endpoint](#heading--10vhalconfig).
+
+Example return value for `curl -s -X GET --unix-socket /run/user/1000/anbox/sockets/api.unix s/1.0/vhal/0x15600503/0x31 | jq .`:
+
+```bash
+{
+  "metadata": {
+    "value": {
+      "area_id": 49,         # Requested area.
+      "bytes": [],           # Raw bytes value as an 8-bit unsigned integer array.
```

Comment:
Same question I left on the MP,  do we need to keep the field when its value is empty? Same for int32_values/int64_values and others.

Reply:
I will rework serialization in a follow-up MP.
For now, I prefer to have the documentation reflect what's actually in the code: if we omit the fields in the future, I will update the documentation and remove them (or explain that they may be omitted).

Reply:
That sounds good to me .